kafka/consumer_handler.go

```diff
@@ -0,0 +1,91 @@
+package kafka
+
+import (
+	"github.com/Financial-Times/go-logger/v2"
+	"github.com/Shopify/sarama"
+	"sync"
+)
+
+// ConsumerHandler represents a Sarama consumer group consumer.
+type ConsumerHandler struct {
+	sync.RWMutex
```

Comment:
This is a bit tricky. The test runner caught a race condition regarding this channel. I believe because there are reads, writes and the channel is getting closed in different goroutines. We can take a look together if you'd like.